tests/integration/metrics_test.go

```diff
@@ -228,3 +234,160 @@ func TestExitedContainersWithoutTTL(t *testing.T) {
 		assert.Empty(t, sample)
 	})
 }
+
+const relativePathToTestdataFileystem = "testdata/cgroupsV1_host/"
+
+func TestAllMetricsPresent(t *testing.T) {
+	expectedSample := biz.Sample{
+		Pids: biz.Pids{
+			Current: pid,
+			Limit:   8000,
+		},
+		Network: biz.Network{
+			RxBytes:   3402,
+			RxDropped: 2,
+			RxErrors:  10,
+			RxPackets: 28,
+			TxBytes:   5,
+			TxDropped: 2,
+			TxErrors:  3,
+			TxPackets: 4,
+		},
+		BlkIO: biz.BlkIO{
+			TotalReadCount:  39,
+			TotalWriteCount: 89,
+			TotalReadBytes:  2387968,
+			TotalWriteBytes: 50,
+		},
+		CPU: biz.CPU{
+			CPUPercent:       0.15003804623431827,
+			KernelPercent:    1.19999999,
+			UserPercent:      100,
+			UsedCores:        2.8546433726,
+			LimitCores:       float64(runtime.NumCPU()),
+			UsedCoresPercent: float64(100) * 2.8546433726 / float64(runtime.NumCPU()),
+			ThrottlePeriods:  2384,
+			ThrottledTimeMS:  96.578349164,
+			Shares:           1024,
+		},
+		Memory: biz.Memory{
+			UsageBytes:            11620352,
+			CacheUsageBytes:       2310144,
+			RSSUsageBytes:         11620352,
+			MemLimitBytes:         104857600,
+			UsagePercent:          11.08203125,
+			KernelUsageBytes:      724992,
+			SwapUsageBytes:        1620352,
+			SwapOnlyUsageBytes:    18446744073699551616,
+			SwapLimitBytes:        209715200,
+			SwapLimitUsagePercent: 0.77264404296875,
+			SoftLimitBytes:        262144000,
+		},
+		RestartCount: 2,
+	}
+
+	// Create a tempDir that will be the root of our mocked filsystem
+	hostRoot := t.TempDir()
+
+	// Create our mocked container cgroups filesystem in the tempDir directory
+	// that will be a symLink to our cgroups testdata
+	cgroupsFolder := filepath.Join(hostRoot, "cgroup")
+	mockFilesystem, err := filepath.Abs(filepath.Join(relativePathToTestdataFileystem, "cgroup"))
+	require.NoError(t, err)
+
+	err = os.Symlink(mockFilesystem, cgroupsFolder)
+	require.NoError(t, err)
+
+	// Create mocked directory proc
+	err = os.Mkdir(filepath.Join(hostRoot, "proc"), 0755)
+	require.NoError(t, err)
+
+	err = generateMockedProcMountsFile(cgroupsFolder, hostRoot)
+	require.NoError(t, err)
+
+	err = generateMockedProcPIDCGroupFile(hostRoot)
+	require.NoError(t, err)
+
+	err = genereateMockedProcNetDevFile(hostRoot)
+	require.NoError(t, err)
+
+	// CgroupsFetcherMock is the raw CgroupsFetcher with mocked cpu.systemUsage and time
+	// The hostRoot is our mocked filesystem
+	cgroupFetcher, err := NewCgroupsFetcherMock(hostRoot, "cgroupfs", "")
+	require.NoError(t, err)
+
+	storer := generateInMemoryStorerWithPreviousCPUState()
+
+	t.Run("Given a mockedFilesystem and previous CPU state Then processed metrics are as expected", func(t *testing.T) {
```

Comment:
I just added it to to have a description of the test being done, it can be removed, we don't have a Standard for the tests. If this would be a Table Test there would be one Run for each execution. I used to do them with different in Run blocks to have a description and actually having all the setup on a Given Block, with the execution on a When Block and the assertions on a Then Block. But here I just wanted to have here the description of the test to not have a superLong Function name.

Reply:
👍 thanks for the explanation! I think it can be a good way to give some context.
As there are plenty of ways to do go tests, I just wanted to know if we are always grouping tests like this or something like that.
As you said, another alternative is having a longer function name and maybe using a description in the `assert` call, but I also like this approach.